pennylane/optimize/shot_adaptive.py

```diff
@@ -347,7 +347,7 @@ def _single_shot_qnode_gradients(self, qnode, args, kwargs):
             self.check_learning_rate(1)
 
         try:
-            device.shots = [(1, self.max_shots)]
+            device.shots = [(1, int(self.max_shots))]

```

Comment:
what were these changes due to?

Reply:
The `valid_int` and `valid_tuple` functions in `shots.py` check to see if the shots are integers, and sometimes `max_shots` is cast as an `np.int64` which causes the checks to fail and raise an error.